libqtile/bar.py

```diff
@@ -308,6 +308,10 @@ def _configure(self, qtile, screen, reconfigure=False):
             )
             self.qtile.renamed_widgets.clear()
 
+        if self.qtile.core.name == "x11":
+            hook.subscribe.setgroup(self.keep_below)
+            hook.subscribe.startup_complete(self.keep_below)
```

Comment:
How come we need to hook to these (i.e. versus doing `self.keep_below` once here only)?

Reply:
Right, but doesn't `keep_below` set a state that will always make it get stacked below visible windows until the flag is removed? Wouldn't the issue you describe only happen if we were moving the window to the bottom as a one-shot change (rather than setting a flag)?

Reply:
The flag does persist and it will mean that the bar is in a lower group than "normal" windows. That bit is True. However, when a window is stacked, the code isn't checking _every_ window each time. It tries to position itself in the correct order within its own layer. It only checks that that layer is correctly placed if there are no other windows in the layer (this is what happens at line 1130 onwards). The theory here is that, when there's only one window in that layer then it's important to make sure that layer is positioned correctly. Any new windows added to the layer don't need the same process as the layer should have already been positioned correctly.

Reply:
OK, gotcha. Then no worries, if having that here makes things easier then +1 from here. I shouldn't let my needless perfectionism get in the way of actual progress :smile: